### PR TITLE
[RD2] Send productMetadata updates non-destructively

### PR DESCRIPTION
**Pull request by bdvorachek**
WI: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07AH000000NXxiYAG/view

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# Features Intended for Future Release

# Features for Elevate Customers

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/classes/UTIL_Http_TEST.cls [hunk 1]
```diff
@@ -307,13 +306,4 @@ public with sharing class UTIL_Http_TEST {
         Test.setMock(HttpCalloutMock.class, callout);
     }
 
-    public static void mockChargeElevateBatch(String responseBody){
-        CalloutMock callout = new CalloutMock()
-                .withStatusCode(UTIL_Http.STATUS_CODE_OK)
```
Comment (lparrott) [inline src/classes/UTIL_Http_TEST.cls:312]: I like the move to a single method, were these Status values just never being checked?

File: src/classes/PS_CommitmentRequest.cls [hunk 2]
```diff
@@ -239,20 +265,40 @@ public inherited sharing class PS_CommitmentRequest {
...
+
+        Map<String, Object> commitment = (Map<String, Object>) responseBody.get('commitment');
+        Map<String, Object> productMetadata = (Map<String, Object>) commitment.get('productMetadata');
+
+        return productMetadata;
+    }
+
+    private Map<String, Object> getElevateCommitment(String commitmentId) {
+        HttpRequest commitmentGetRequest = buildGetRequest(commitmentId);
+        UTIL_Http.Response response = requestService.sendRequest(commitmentGetRequest);
```
Comment (lparrott) [inline src/classes/PS_CommitmentRequest.cls:296]: Not sure where to add it, but I'd feel better if this was in a try/catch 😬

**Review by lparrott — COMMENTED** [review]
Looking great, you even got in some good cleanup and refactoring 👍 Just had one comment about a potential try/catch block.

File: src/classes/PS_CommitmentRequest.cls (hunk 2 above)
Comment (bdvorachek) [reply src/classes/PS_CommitmentRequest.cls:296]: The same try/catch that's in the update call would catch any errors from this callout, which I think is what we'd want. If the GET request times out, we'd want the same handling as a PUT/POST timeout. 
https://github.com/SalesforceFoundation/NPSP/blob/41a6bcd22c75e9c720cbfb4cf6b6d04ca1c1792b/src/classes/RD2_EntryFormController.cls#L278

File: src/classes/UTIL_Http_TEST.cls (hunk 1 above)
Comment (bdvorachek) [reply src/classes/UTIL_Http_TEST.cls:312]: These status values are the default values in the constructor.

File: src/classes/PS_CommitmentRequest.cls (hunk 2 above)
Comment (lparrott) [reply src/classes/PS_CommitmentRequest.cls:296]: Oh, excellent, I should've looked further upstream 👍

**Review by lparrott — APPROVED** [review]

**Review by npsp-reedestockton — APPROVED** [review]
This is excellent, @bdvorachek!!!  Very clean code!  It should be very easy to extend this as needed.

**Review by dfuller114 — APPROVED** [review]
looks good! thanks Ben!

**Pull request merged**